registry/registry.go

```diff
@@ -2,68 +2,70 @@
 package registry
 
 import (
-	"context"
-	"encoding/base64"
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"net/http"
-	"net/http/cookiejar"
-	"sort"
-	"strconv"
-	"strings"
-	"time"
-
 	"github.com/go-kit/kit/log"
-	dockerregistry "github.com/heroku/docker-registry-client/registry"
-	"github.com/jonboulle/clockwork"
-	"golang.org/x/net/publicsuffix"
-
 	"github.com/weaveworks/flux"
-	fluxmetrics "github.com/weaveworks/flux/metrics"
+	"sort"
+	"time"
 )
 
 const (
-	dockerHubHost    = "index.docker.io"
-	dockerHubLibrary = "library"
-
 	requestTimeout = 10 * time.Second
 )
 
-type creds struct {
-	username, password string
+// Old registry interface. Used as an adaptor to the new class
+type Client interface {
+	GetRepository(repository string) ([]flux.ImageDescription, error)
```

Comment:
For the moment, yes. This is an adapter between the old code and the new one. We can remove it shortly.